### PR TITLE
disabled switch to change from free to paid event if an event has reg…

### PR DESCRIPTION
**Pull request by MadsNyl**
…istrations

## Description

This refactor will fix the follwoing issues:

- You can now switch from free to paid event (and vice versa) when there are none registered users, but not when there are.
- You will not be creating a payment order until you press the vipps button. Therefore you will get a new order every time you press the button. This will prevent session timeouts.
- When you unregister from an event after you have paid, your registration will be deleted and you will get refunded your money. This will be notified in the snackbar, and in notifications and with mail. Therefore you will have to pay again if you register again.

closes #

Changes:

Screenshots:

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] The PR includes a picture showing visual changes
- [ ] Added Analytics-events if relevant
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)

### CODE REVIEW

File: src/pages/EventAdministration/components/EventEditor.tsx [hunk 1]
```diff
@@ -476,6 +476,7 @@ const EventEditor = ({ eventId, goToEvent }: EventEditorProps) => {
             }
             name='is_paid_event'
             type='switch'
+            disabled={data?.list_count ? data.list_count > 0 : false}
```
Comment (martcl) [inline src/pages/EventAdministration/components/EventEditor.tsx:479]: > Props should be sorted alphabetically

I tillegg:

```suggestion
            disabled={data?.list_count ? data.list_count > 0}
```

**Review by martcl — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@
 ## Neste versjon
 - ✨ **Bannere**. Tidsbegrensede bannere kan nå opprettes.
 
+## versjon 2023.17.10
```
Comment (martcl) [inline CHANGELOG.md:16]: Ikke skriv versjon selv. Det fikses når vi lanserer i prod

File: src/pages/EventAdministration/components/EventEditor.tsx [hunk 3]
```diff
@@ -467,6 +467,7 @@ const EventEditor = ({ eventId, goToEvent }: EventEditorProps) => {
           </Row>
           <Bool
             control={control}
+            disabled={!(data?.list_count && data?.list_count > 0)}
```
Comment (martcl) [inline src/pages/EventAdministration/components/EventEditor.tsx:470]: Jeg tror dette fungerer altså, men kanskje det er noe typescript wodo som foregår
```suggestion
            disabled={data?.list_count && data.list_count > 0}
```

**Review by martcl — CHANGES_REQUESTED** [review]

File: src/pages/EventAdministration/components/EventEditor.tsx (hunk 3 above)
Comment (MadsNyl) [reply src/pages/EventAdministration/components/EventEditor.tsx:470]: Ja funker nok, men TS klager på det hvis jeg ikke tar ! foran

Comment (martcl) [reply src/pages/EventAdministration/components/EventEditor.tsx:470]: Det å ta utropstegn forran gjør akkuratt motsatt av det vi ønsker. Nå vil den være disabled når vi ikke har deltagere, mens enabled når vi har deltagere

Comment (MadsNyl) [reply src/pages/EventAdministration/components/EventEditor.tsx]: Jupp har fiksa nå

File: CHANGELOG.md (hunk 2 above)
Comment (MadsNyl) [reply CHANGELOG.md]: Fikset nå

**Pull request merged**